src/gurt/telemetry.c

```diff
@@ -1916,6 +1950,23 @@ int d_tm_add_metric(struct d_tm_node_t **node, int metric_type, char *sh_desc,
 	if (fmt == NULL)
 		return -DER_INVAL;
 
+	if (units != NULL) {
+		switch (metric_type & D_TM_ALL_NODES) {
+		case D_TM_TIMESTAMP:
+			return -DER_INVAL;
+		case D_TM_TIMER_SNAPSHOT:
+			if ((strncmp(units, D_TM_MICROSECOND,
+				     D_TM_MAX_UNIT_LEN)) != 0)
+				return -DER_INVAL;
+		case D_TM_DURATION:
+			if ((strncmp(units, D_TM_MICROSECOND,
+				     D_TM_MAX_UNIT_LEN)) != 0)
+				return -DER_INVAL;
```

Comment:
Yes, I see that's a bit confusing.  I'll rework it so that the units are set automatically inside d_tm_add_metric() for those metrics where it makes sense to do so.